### ISSUE TITLE
error: package ID specification `arrow_ws` did not match any packages

### ISSUE THREAD

**Issue by mlilback** [intellij-rust/intellij-rust#10190]
## Environment

* **IntelliJ Rust plugin version:** 0.4.185.5086-222
* **Rust toolchain version:** 1.69.0-nightly (d962ea578 2023-02-26) x86_64-apple-darwin
* **IDE name and version:** IntelliJ IDEA 2022.2.4 Ultimate Edition (IU-222.4459.24)
* **Operating system:** macOS 12.6.1
* **Macro expansion:** enabled

## Problem description
I created a branch off main. I did work, adding a new module named `arrow_ws`. I then need to work on another feature so I created a new branch from main. When building in intellij, I get the error `error: package ID specification 'arrow_ws' did not match any packages`. The project builds fine from the command line.

## Steps to reproduce
1. Create a project and commit to git
2. Create a branch and add a new module. Compile, run, commit.
3. Create a new branch from the same commit where the first branch came from.
4. Try to build. You will get the error described above.

**Comment by fschwa** [comment]
I got the same error message by just renaming the package in Cargo.toml.
Also same result: build on command line, does not build in CLion.

**Comment by fschwa** [comment]
I guess the problem is an .idea folder not matching the code.
In my case the Run config still had the old project name. So I deleted the Run config, created a new one and it worked.

In your case, I guess the .idea folder is not in version control? That way it "remembers" the newly added module over "branch barriers". So it expects the module to be there in a branch where the module is not there.

But I am new to Rust, so maybe I am mistaken.

**Comment by grosubogdan** [comment]
I got the same and is not related to .idea folder. There is no where specified in my folder the old name.

**Comment by mwmuni** [comment]
Solution is to change the name of the package in the .idea folder and adjust the references to that file. By way of example:

Assuming we want to rename a project from `ApplePie` to `apple_pie`...

Rename: `$PROJECT_DIR$/.idea/ApplePie.iml` -> `$PROJECT_DIR$/.idea/apple_pie.iml`
Modify:
```xml
file: $PROJECT_DIR$/.idea/modules.xml
...
<module fileurl="file://$PROJECT_DIR$/.idea/ApplePie.iml" filepath="$PROJECT_DIR$/.idea/ApplePie.iml" />
...
```
Change the above line to:
```xml
<module fileurl="file://$PROJECT_DIR$/.idea/apple_pie.iml" filepath="$PROJECT_DIR$/.idea/apple_pie.iml" />
```

Last but not least, change your run configuration from:
```bash
run --package ApplePie --bin ApplePie
```
to
```bash
run --package apple_pie --bin apple_pie
```

**Comment by osmundf** [comment]
Still an issue in RustRover 2024.1 EAP Build #RR-241.15989.180, built on May 6, 2024

**Comment by pprogrammingg** [comment]
yeah I have the same issue, even after following the instruction below.

Build #RR-241.15989.199, built on May 18, 2024



> Solution is to change the name of the package in the .idea folder, adjust the references to that file and finally change the run configuration to use the new package. By way of example:
> 
> Assuming we want to rename a package from `ApplePie` to `apple_pie`...
> 
> Rename: `$PROJECT_DIR$/.idea/ApplePie.iml` -> `$PROJECT_DIR$/.idea/apple_pie.iml` Modify:
> 
> ```
> file: $PROJECT_DIR$/.idea/modules.xml
> ...
> <module fileurl="file://$PROJECT_DIR$/.idea/ApplePie.iml" filepath="$PROJECT_DIR$/.idea/ApplePie.iml" />
> ...
> ```
> 
> Change the above line to:
> 
> ```
> <module fileurl="file://$PROJECT_DIR$/.idea/apple_pie.iml" filepath="$PROJECT_DIR$/.idea/apple_pie.iml" />
> ```
> 
> Last but not least, change your run configuration from:
> 
> ```shell
> run --package ApplePie --bin ApplePie
> ```
> 
> to
> 
> ```shell
> run --package apple_pie --bin apple_pie
> ```
